src/ch02-02-functions.md

```diff
@@ -0,0 +1,164 @@
+## Functions
+
+Together with `struct`, functions are the core building blocks of hot loading in
+Mun. Throughout this documentation you've already seen a lot of examples of
+functions. You have also seen the `fn` keyword, which allows you to define a
+function.
+
+Mun uses *snake case* as the conventional style for function and variable names.
+In snake case all letters are lowercase and words are separated by underscores. 
+
+```mun
+fn main() {
+    another_function();
+}
+
+fn another_function() {
+
+}
+```
+
+Function definitions in Mun start with an optional access modifier (`pub`)
+followed by the `fn` keyword, a name, an argument list enclosed by parentheses,
+an optional return type specifier, and finally a body. 
+
```

Comment:
```suggestion

### Function Access Modifier

Marking a function with the `pub` keyword allows you to use it from outside of
the module it is defined in.

```mun
// This function is not accessible outside of this code
fn foo() {
    // ...
}
// This function is accessible from anywhere.
pub fn bar() {
    // Because `bar` and `foo` are in the same file, this call is valid.
    foo()
}```

When you want to interface from your host language (C++, Rust, etc.) with Mun,
you can only access `pub` functions. These functions are hot reloaded by the
runtime when they **or** functions they call have been modified.

```